pycodestyle

ssssarah opened the pull request:
Inspired by https://github.com/BlueBrain/nexus-forge/pull/237 , but on the more recent codebase.

ssssarah committed:
pycodestyle

ssssarah committed:
pylint

ssssarah commented on .pylintrc:
Disabled docstring warnings because there are many of them, but in the future these should be re-enabled and docstrings should be written

ssssarah replied:
- W0511: TODO-FIXME
- W0123: use of eval
- W0212: Access to protected member
- W0246: redefinition of constructor in subclass, but only calling super (nothing else added)

ssssarah replied:
(I added some ignoring in order to have a better view of the warnings I can easily address but I don't actually believe these should be ignored)

MFSY commented:
> Inspired by #237 , but on the more recent codebase. Running `tox -e lint` led to many warnings. This is just addressing some of them. Maybe running the command should be done as job systematically (as proposed in the previously mentioned MR).

Please add the job.

ssssarah committed:
more linting

ssssarah committed:
created job for pycodestyle using tox, disabled pylinting until fixed

ssssarah committed:
rm ignored pycodestyle warnings

ssssarah commented:
> > Inspired by #237 , but on the more recent codebase. Running `tox -e lint` led to many warnings. This is just addressing some of them. Maybe running the command should be done as job systematically (as proposed in the previously mentioned MR).
> 
> Please add the job.

I've just added the job in the github workflows, but it only runs pycodestyle because I've fixed all pycodestyle errors, but I haven't fixed all pylint errors, so if added the pylint one it would fail

ssssarah committed:
more linting

ssssarah committed:
fix

ssssarah committed:
linting

crisely09 commented:
It's fine with me. I will approve but let @MFSY have the last word.

ssssarah committed:
fix all import order

ssssarah committed:
more linting

ssssarah committed:
test lint change

ssssarah committed:
more linting

ssssarah committed:
reenable pylint, fail under 9.0 score

ssssarah committed:
lower threshold for pylint success to 8.5

ssssarah commented:
Although I don't think it helps much, I've re-enabled pylint and added a threshold under which the job fails. The score should be above 8.5/10
